packages/insured-bridge-relayer/src/CrossDomainFinalizer.ts

```diff
@@ -230,9 +233,15 @@ export class CrossDomainFinalizer {
     // Note that the query below only works on particular RPC endpoints. Infura, for example, is limited to a 100k look
     // back. This means that to use this module you need to use an endpoint that supports longer lookbacks, such as
     // alchemy, which supports arbitrary long loobacks. In future, Infura will support arbitrary long lookbacks.
+    // Note2: the blockOffset is added to limit how far forward we look. We want to ideally ignore the newest blocks
+    // as these can causes errors in the Bridge Adapters if the L2 transactions are not yet checkpointed on the
+    // canonical L1 StateCommitmentChain. 5000 blocks is ~ 3 hours at current Optimism rate. This is only applied if we
+    // are not running in test mode.
+    const blockOffset =
```

Comment:
I think it'd be better to set this offset via an environment variable. The default value could be 5000 blocks if the bot is being run in prod or 0 in test

Reply:
Or we could put the constant value in the ask so the front end would be able to use it also

Reply:
Not sure why this would ever be needed in the FE, but I think it does make sense to put in the SDK!

Agree on the env variable, but going to go ahead and merge -- we can add that later as necessary.